### ISSUE TITLE
Filters Design and Development

### ISSUE THREAD

**Issue by Xxxuan11** [RTEP-zero-to-one/FacialDecoration#10]
Applied to video streams, allowing users to adjust various parameters according to their preferences

**Comment by XeonHis** [comment]
DARKTONE filter has some problem. Temporary discorded. @SheenaWang11 

Error Message: 
```
C++ exception with description "OpenCV(4.5.5) C:\opencv\sources\modules\core\src\matmul.dispatch.cpp:356: error: (-215:Assertion failed) (type == (((5) & ((1 << 3) - 1)) + (((1)-1) << 3)) || type == (((6) & ((1 << 3) - 1)) + (((1)-1) << 3)) || type == (((5) & ((1 << 3) - 1)) + (((2)-1) << 3)) || type == (((6) & ((1 << 3) - 1)) + (((2)-1) << 3))) in function 'gemm'
" thrown in the test body.
```

Problem reproduct by follow:
1. git checkout 10-filters-design-and-development (SHA: 02ec9ae5ac36ae29c28c219681c12088c20941ad)
2. uncomment line 89-94 in filter_process.cpp and line 62-70 in test_function.cpp
3. run TEST(OpenCVTest, filter_DARKTONE)

**Comment by XeonHis** [comment]
filter COMICBOOK crashes in realtime process.

Reproduction:

1. checkout in branch 13-filter-add-in-realtime-video-and-switch-by-click-in-gui (SHA: 9008544a60d4a8e78110734945c4c459d739f9e7)
2. enable button COMICBOOK
3. run GUI -> start video -> click COMICBOOK filter

**Comment by XeonHis** [comment]
Problem fixed by cvtcolor convert 8UC1 to 8UC3.
Fix in GUI branch c25bb612ec81fbf2169645549ede670cbb07b78b
